data_diff/diff_tables.py

```diff
@@ -148,6 +150,13 @@ def diff_sets(a: set, b: set) -> iter:
 
 DiffResult = iter  # Iterator[Tuple[Literal["+", "-"], tuple]]
 
+def thread_map(func, iter):
+    task_pool = ThreadPoolExecutor()
+    return task_pool.map(func, iter)
+
+def precalc_attr(attr, iter):

```

Comment:
`precalc_` doesn't imply to me it's threaded, is this naming a Python pattern I am not aware of? Could we call it `threaded_map_attr` or something?

Reply:
I like threading in it if you don't mind

Reply:
done